### PR TITLE
Issues #23, #29, #25, #26, #27: dealing with inputs

### PR DESCRIPTION
**Pull request by Clemcarle**
Added inputs and optimal output as member variables of environment. 
Added a function to change the inputs by drawing n new values from the environment cue distribution. 
The inputs from the environment can be used in the simulation.

### CODE REVIEW

File: environment.cpp [hunk 1]
```diff
@@ -92,7 +96,8 @@ std::vector<double> create_n_inputs(environment e, const int &n_inputs, std::mt1
     return input_vector;
 }
 
-std::vector<double> create_n_inputs(std::uniform_real_distribution<double> dist, const int &n_inputs, std::mt19937_64 &rng)
+std::vector<double> create_n_inputs(std::uniform_real_distribution<double> dist,
+                                    const int &n_inputs, std::mt19937_64 &rng)
```
Comment (swom) [inline environment.cpp]: 👍 for the indent

**Review by swom — COMMENTED** [review]
Gut I will merge and solve the conflicts

File: simulation.cpp [hunk 2]
```diff
@@ -236,6 +242,12 @@ std::vector<double> new_inputs = create_n_inputs(s.get_env(), get_current_input(
 assign_inputs(s.get_pop(), new_inputs);
 }
 
+void create_inputs(simulation &s)
+{
+  environment &e = s.get_env();
+  e.update_n_inputs(s.get_rng(), s.get_inds_input_size());
```
Comment (swom) [inline simulation.cpp:248]: gut, I still feel there is a safer way to do this. Here we are playing with fire a little bit

**Comment by swom** [comment]
MMMM nope, ok, here we have a tad too many merge conflicts. I am wondering why this is so. Did you remember to merge develop constantly to your brnach while you were working on it? This is usually the cause of merge conflicts, develop and the branch go out of sync. I think this will take me awhile to solve. I will merge once things are fixed.

File: simulation.cpp (hunk 2 above)
Comment (Clemcarle) [reply simulation.cpp:248]: I am not sure I understand why?

**Comment by Clemcarle** [comment]
I think the merge conflicts are because of working on different branches and then merging them to develop at different times. I can solve it, I have everything fresh in mind.

**Comment by swom** [comment]
> I think the merge conflicts are because of working on different branches and then merging them to develop at different times. I can solve it, I have everything fresh in mind.

I will try to fix it, I just wnat to make sure I understand what is going on.

**Pull request merged**